core/app/models/concerns/spree/user_reporting.rb

```diff
@@ -3,20 +3,31 @@ module UserReporting
     extend DisplayMoney
     money_methods :lifetime_value, :average_order_value
 
-    def lifetime_value
-      orders.complete.sum(:total)
+    def lifetime_value(*args)
+      order_calculate(*args,
+                      operation: :sum,
+                      column: :total)
     end
 
-    def order_count
-      orders.complete.size
+    def average_order_value(*args)
+      order_calculate(*args,
+                      operation: :average,
+                      column: :total)
     end
 
-    def average_order_value
-      if order_count > 0
-        lifetime_value / order_count
-      else
-        BigDecimal('0.00')
-      end
+    def order_count(store = nil)
+      order_calculate(store,
+                      store.supported_currencies.split(','),
```

Comment:
```suggestion
                      store&.supported_currencies&.split(','),
```

Reply:
@szymoniwacz rather than doing this we should fallback `store` to the default one, eg

```ruby
store ||= Store.default
```

Reply:
@damianlegawiec @szymoniwacz changed to `store ||= Store.default`